src/cargo/util/command_prelude.rs

```diff
@@ -492,6 +499,7 @@ pub trait ArgMatchesExt {
             target_rustc_args: None,
             local_rustdoc_args: None,
             rustdoc_document_private_items: false,
+            honor_rust_version: !self._is_present("ignore-rust-version"),
```

Comment:
This should probably generate an error if it is not on nightly.

Reply:
I'm not sure how to do that -- any hints?

Reply:
Fixed.